src/python_testing/TC_LCFG_2_1.py

```diff
@@ -0,0 +1,158 @@
+            endpoint=endpoint
+        )
+
+        # Verify values in SupportedLocales attribute are not repeated
+        asserts.assert_false(self.has_repeated_values(initial_values_supported_locales),
+                             "SupportedLocales attribute has no repeated values")
+
+        # Verify maximun number of elements in the SupportedLocales list is 32
+        asserts.assert_true(len(initial_values_supported_locales) <= 32, "SupportedLocales attribute has less than 32 elements")
+
+        # Verify values of SupportedLocales attribute has a maximum lenght of 35 bytes
+        asserts.assert_true(self.list_has_maximum_length_of_35_bytes(initial_values_supported_locales),
+                            "SupportedLocales attribute has a maximum lenght of 35 bytes")
+
+        # Step 2: TH reads ActiveLocale attribute from the DUT
+        self.step(2)
+
+        initial_active_locale = await self.read_single_attribute_check_success(
+            cluster=Clusters.LocalizationConfiguration,
+            attribute=Clusters.LocalizationConfiguration.Attributes.ActiveLocale,
+            endpoint=endpoint
+        )
+
+        # Verify that the ActiveLocale attribute is not empty
+        asserts.assert_true(bool(initial_active_locale), "ActiveLocale attribute is not empty")
+
+        # Verify that the ActiveLocale attribute is Language Tag as defined by BCP47
+        asserts.assert_true(langcodes.tag_is_valid(initial_active_locale),
+                            "ActiveLocale attribute is Language Tag as defined by BCP47")
+
+        # Verify that the value of ActiveLocale attribute has maximum lenght of 35 bytes
+        asserts.assert_true(self.has_maximum_lenght_of_35_bytes(initial_active_locale),
+                            "ActiveLocale attribute has less than 35 bytes")
+
+        # Verify that the ActiveLocale attribute value is present in the SupportedLocales attribute list
+        asserts.assert_true(self.supported_locales_has_active_locale(initial_values_supported_locales,
+                            initial_active_locale), "ActiveLocale attribute value is present in the SupportedLocales attribute list")
+
+        # Step 3: TH writes new string not present in SupportedLocales attribute to ActiveLocale attribute
+        self.step(3)
+
+        result = await self.write_single_attribute(attribute_value=Clusters.LocalizationConfiguration.Attributes.ActiveLocale(value_not_present_in_supported_locales), endpoint_id=endpoint, expect_success=False)
+
+        # DUT responds with CONTRAINT_ERROR
+        asserts.assert_equal(result, Status.ConstraintError, "Trying to write an invalid value should return ConstraintError")
+
+        # Step 4: TH writes new string present in SupportedLocales attribute to ActiveALocale attribute
+        self.step(4)
+
+        value_present_in_supported_locales = initial_values_supported_locales[initial_values_supported_locales.index(es_ES)]
```

Comment:
this should not be hardcoded, but should come from the supported locales list. The list is not static, so this will not necessarily appear on every device.